open-metadata-implementation/access-services/subject-area/subject-area-server/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/handlers/SubjectAreaCategoryHandler.java

```diff
@@ -423,4 +412,95 @@ private void updateAttributes(Category currentCategory, Category newTerm) {
 
         return response;
     }
+
+    /**
+     * Get the terms that are categorized by this Category
+     *
+     * @param userId       unique identifier for requesting user, under which the request is performed
+     * @param guid         guid of the category to get terms
+     * @param termHandler term handler
+     * @param startingFrom initial position in the stored list.
+     * @param pageSize     maximum number of definitions to return on this call.
+     * @return A list of terms is categorized by this Category
+     * when not successful the following Exception responses can occur
+     * <ul>
+     * <li> UserNotAuthorizedException           the requesting user is not authorized to issue this request.</li>
+     * <li> InvalidParameterException            one of the parameters is null or invalid.</li>
+     * <li> PropertyServerException              Property server exception. </li>
+     * </ul>
+     * */
+    public SubjectAreaOMASAPIResponse<Term> getCategorizedTerms(String userId, String guid, SubjectAreaTermHandler termHandler, int startingFrom, Integer pageSize) {
```

Comment:
I really don't like how methods like this are written. But, unfortunately, I do not have time to show you how I would have done it ...
I am very sorry about this ...

Reply:
I have to approve this because I have no free time right now, but for you this is an urgent change.

Reply:
Thanks for your understanding @immmus. I am interested in how you would prefer the methods to be written.